hordes-ui-mod.js

```diff
@@ -382,941 +460,1071 @@
+            $tab.classList.add('js-selected-tab');
+
+            const tabFilters = state.chatTabs.find(tab => tab.id === tabId).filters;
+            // Simulating clicks on the filters to turn them on/off
+            const $filterButtons = Array.from(document.querySelectorAll('.channelselect small'));
+            Object.keys(tabFilters).forEach(filter => {
+                const $filterButton = $filterButtons.find($btn => $btn.textContent === filter);
+                const isCurrentlyFiltered = $filterButton.classList.contains('textgrey');
+
+                // If is currently filtered but filter for this tab is turned off, click it to turn filter off
+                if (isCurrentlyFiltered && !tabFilters[filter]) {
+                    $filterButton.click();
+                }
+                // If it is not currently filtered but filter for this tab is turned on, click it to turn filter on
+                if (!isCurrentlyFiltered && tabFilters[filter]) {
+                    $filterButton.click();
+                }
+            });
+
+            // Update state for our custom chat filters to match the tab's configuration, then filter chat for it
+            const isGMChatVisible = !tabFilters.GM;
+            modHelpers.setGMChatVisibility(isGMChatVisible);
+
+            // Update the selected tab in state
+            state.selectedChatTabId = tabId;
+            save();
+        },
+
+        // Updates state.chat.GM and the DOM to make text white/grey depending on if gm chat is visible/filtered
+        // Then filters chat and saves updated chat state
+        setGMChatVisibility: isGMChatVisible => {
+            const $chatGM = document.querySelector(`.js-chat-gm`);
+            state.chat.GM = isGMChatVisible;
+            $chatGM.classList.toggle('textgrey', !state.chat.GM);
+            modHelpers.filterAllChat();
+            save();
+        },
+
+        getCurrentCharacterLvl: () => Number(document.querySelector('#ufplayer .bgmana > .left').textContent.split('Lv. ')[1]),
+
+        getCurrentXp: () => Number(document.querySelector('#expbar .progressBar > .left').textContent.split('/')[0].trim()),
+
+        getNextLevelXp: () => Number(document.querySelector('#expbar .progressBar > .left').textContent.split('/')[1].trim()),
+
+        /** user invoked reset of xp meter stats */
+        resetXpMeterState: () => {
+            state.xpMeterState.xpGains = []; /** array of xp deltas every second */
+            state.xpMeterState.averageXp = 0;
+            state.xpMeterState.gainedXp = 0;
+            document.querySelector('.js-xp-time').textContent = '-:-:-';
+        },
+
+        /** toggle the xp meter */
+        toggleXpMeterVisibility: () => {
+            const xpMeterContainer = document.querySelector('.js-xpmeter');
+            xpMeterContainer.style.display === 'none' ? xpMeterContainer.style.display = 'block' : xpMeterContainer.style.display = 'none';
```

Comment:
We can shorten this to `xpMeterContainer.style.display = xpMeterContainer.style.display === 'none' ? 'block' : 'none';`